server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -350,6 +351,9 @@ protected Node(
             final ClusterService clusterService = new ClusterService(settings, settingsModule.getClusterSettings(), threadPool);
             clusterService.addStateApplier(scriptModule.getScriptService());
             resourcesToClose.add(clusterService);
+            clusterService.addLocalNodeMasterListener(
+                    new ConsistentSettingsService(settings, clusterService, settingsModule.getConsistentSecureSettings())
+                            .newHashPublisher());
```

Comment:
This is how the publisher of secure settings is registered.
In the "reloadable" use case - which is not targeted in this PR - the `settings` object would need to be dynamic with respect to the `SecureSettings` contents, rather than creating a new one. Otherwise, the published hashes would be those of the original contents of the secret store, when the node booted.